test/src/tests-batchUploader.ts

```diff
@@ -407,13 +408,12 @@ describe('batch uploader', () => {
             consentState.setCCPAConsentState(ccpaConsent);
             consentState.addGDPRConsentState('test purpose', gdprConsent);
 
-            // Mock the query params capture function because we cannot mock window.location.href
-            sinon.stub(window.mParticle.getInstance()._IntegrationCapture, 'getQueryParams').returns({
-                rtid: 'test-click-id',
-            });
-
             window.mParticle.init(apiKey, window.mParticle.config);
             await waitForCondition(hasIdentifyReturned);
+            const integrationCapture = window.mParticle.getInstance()._IntegrationCapture;
+            // Mock the query params capture function because we cannot mock window.location.href
+            sinon.stub(integrationCapture, 'getQueryParams').returns({ rtid: 'test-click-id' });
+            integrationCapture.capture();
```

Comment:
doesn't seem like you need this line here.  the capture will happen during the visibility change, which runs through the batchuploader code

(I pulled this down locally and commented it out. but you can try as well)